indexer/services/roundtable/__tests__/tasks/uncross-orderbook.test.ts

```diff
@@ -0,0 +1,132 @@
+  });
+
+  it('succeeds without any crossed levels', async () => {
+    const perpetualMarkets: PerpetualMarketFromDatabase[] = await PerpetualMarketTable.findAll(
+      {},
+      [],
+    );
+    const market = perpetualMarkets[0];
+    await OrderbookLevelsCache.updatePriceLevel({
+      ticker: market.ticker,
+      side: OrderSide.BUY,
+      humanPrice: '30100',
+      sizeDeltaInQuantums: '2000',
+      client: redisClient,
+    });
+    await OrderbookLevelsCache.updatePriceLevel({
+      ticker: market.ticker,
+      side: OrderSide.SELL,
+      humanPrice: '45000',
+      sizeDeltaInQuantums: '1000',
+      client: redisClient,
+    });
+
+    await runTask();
+    expect(logger.info).not.toHaveBeenCalled();
+  });
+
+  it('removes crossed bid and ask levels', async () => {
+    const perpetualMarkets: PerpetualMarketFromDatabase[] = await PerpetualMarketTable.findAll(
+      {},
+      [],
+    );
+    const market = perpetualMarkets[0];
+    await OrderbookLevelsCache.updatePriceLevel({
+      ticker: market.ticker,
+      side: OrderSide.BUY,
+      humanPrice: '45100',
+      sizeDeltaInQuantums: '2000',
+      client: redisClient,
+    });
+    await OrderbookLevelsCache.updatePriceLevel({
+      ticker: market.ticker,
+      side: OrderSide.SELL,
+      humanPrice: '45000',
+      sizeDeltaInQuantums: '1000',
+      client: redisClient,
+    });
+
+    await runTask();
+
+    expect(require('@dydxprotocol-indexer/redis/build/src/caches/orderbook-levels-cache').deleteStalePriceLevel).toHaveBeenCalledWith(expect.objectContaining({
+      side: OrderSide.BUY,
+      humanPrice: '45100',
+    }));
+    expect(require('@dydxprotocol-indexer/redis/build/src/caches/orderbook-levels-cache').deleteStalePriceLevel).not.toHaveBeenCalledWith(expect.objectContaining({
+      side: OrderSide.SELL,
+      humanPrice: '45000',
+    }));
+  });
+
+  it('logs a failure to delete stale bid or ask level', async () => {
+    const { deleteStalePriceLevel } = require('@dydxprotocol-indexer/redis/build/src/caches/orderbook-levels-cache');
+    deleteStalePriceLevel.mockImplementationOnce(() => false);
+
+    const perpetualMarkets: PerpetualMarketFromDatabase[] = await PerpetualMarketTable.findAll(
+      {},
+      [],
+    );
+    const market = perpetualMarkets[0];
+    await OrderbookLevelsCache.updatePriceLevel({
+      ticker: market.ticker,
+      side: OrderSide.BUY,
+      humanPrice: '45100',
+      sizeDeltaInQuantums: '2000',
+      client: redisClient,
+    });
+    await OrderbookLevelsCache.updatePriceLevel({
+      ticker: market.ticker,
+      side: OrderSide.SELL,
+      humanPrice: '45000',
+      sizeDeltaInQuantums: '1000',
+      client: redisClient,
+    });
+
+    await runTask();
+
+    expect(logger.info).toHaveBeenCalledWith(expect.objectContaining({
+      message: expect.stringContaining('Failed to delete stale bid level for'),
+      side: OrderSide.BUY,
+      humanPrice: '45100',
+    }));
+  });
+});
```

Comment:
add a test for deleting multiple crossed price levels?

Reply:
done